src/main/java/org/folio/rest/impl/ItemStorageDereferencedAPI.java

```diff
@@ -0,0 +1,198 @@
+ * Aim is to determine if dereferencing item records
+ * results in performance improvement when performing
+ * checkout/checkin operations.
+ */
+public class ItemStorageDereferencedAPI implements ItemStorageDereferenced {
+  private static final Logger logger = LogManager.getLogger("okapi");
+  public static final String ITEM_TABLE = "item";
+  private static final String JSON_COLUMN = "jsonb";
+  private static Map<String, String> fields = new HashMap<>();
+  private static List<String> idFields = new ArrayList<String>();
+  static{
+    fields.put("instance", "instanceRecord");
+    fields.put("holdings", "holdingsRecord");
+    fields.put("materialtype", "materialType");
+    fields.put("permloan", "permanentLoanType");
+    fields.put("temploan", "temporaryLoanType");
+    fields.put("permloc", "permanentLocation");
+    fields.put("temploc", "temporaryLocation");
+    fields.put("effloc", "effectiveLocation");
+    idFields.add("holdingsRecordId");
+    idFields.add("instanceId");
+    idFields.add("materialTypeId");
+    idFields.add("permanentLocationId");
+    idFields.add("temporaryLocationId");
+    idFields.add("effectiveLocationId");
+    idFields.add("permanentLoanTypeId");
+    idFields.add("temporaryLoanTypeId");
+  }
+  private String sqlQuery= "SELECT item.jsonb as item, holdingstable.jsonb as " + fields.get("holdings") + ",\n" + 
+  "instancetable.jsonb as " + fields.get("instance") + ", materialtypetable.jsonb as " + fields.get("materialtype") + ",\n" + 
+  "locationtable.jsonb as " + fields.get("permloc") + ", locationtabletemp.jsonb as " + fields.get("temploc") + ",\n" + 
+  "locationtableeffective.jsonb as " + fields.get("effloc") + ",\n"+
+  "loantable.jsonb as " + fields.get("permloan") + ", loantabletemp.jsonb as " + fields.get("temploc") + "\n"+
+  "FROM " + ITEM_TABLE + "\n"+
+  "INNER JOIN holdings_record as holdingstable on item.holdingsrecordid=holdingstable.id\n"+
+  "INNER JOIN instance as instancetable on holdingstable.instanceid=instancetable.id\n"+
+  "INNER JOIN material_type as materialtypetable on item.materialtypeid=materialtypetable.id\n"+
+  "LEFT JOIN location as locationtable on item.permanentlocationid = locationtable.id\n"+
+  "LEFT JOIN location as locationtabletemp on item.temporarylocationid = locationtabletemp.id\n"+
+  "LEFT JOIN location as locationtableeffective on item.effectivelocationid = locationtableeffective.id\n"+
+  "INNER JOIN loan_type as loantable on item.permanentloantypeid=loantable.id\n"+
+  "LEFT JOIN loan_type as loantabletemp on item.temporaryloantypeid=loantabletemp.id\n";
+
+  @Validate
+  @Override
+  public void getItemStorageDereferencedItems(
+    int offset, int limit, String query, String lang,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    List<DereferencedItem> mappedResults = new ArrayList<>();
+    String whereClause = "";
+
+    if (query != null) {
```

Comment:
For this particular endpoint, maybe we want to require the CQL query and not allow the entire result set to be fetched by default.

Reply:
Well, it will still default limit to ten records.  I dunno, I tend to want endpoints to conform to established design patterns-and from what I can see, the standard design pattern for these endpoints is "If you don't give me a CQL query, I'll give you everything, starting with the first ten records."

Reply:
Ok, let's leave it as is.